### ISSUE TITLE
Use pytesseract on other lang issue - more clue

### ISSUE THREAD

**Issue by moonlovestar** [madmaze/pytesseract#94]
Hello guys,

    Although I update to pytesseract v0.1.8, it cannot do ocr on JPN image still.
    However, if I manually set command to following, it can works.
   (but program will crash after that because I change the file path. It doesn't matter for now.)

    def run_tesseract(input_filename, ...):
        ...
        if not sys.platform.startswith('win32') and nice != 0:
            command  += ('nice', '-n', str(nice))
        +input_filename = "<my src img path>\\src.png"
        +output_filename_base = "<my src img path>\\output.log"
        command += (tesseract_cmd, input_filename, output_filename_base)
       ...
 
I change src_img path to absolute path for my src file and save the log to absolute location.

In the original code, I cannot get anything from tesseract ocr.
If I change src_img to absolute path without modified output.log part, it will get part of result.(some jpn word have lost.)

If I cange src_img and modified output together, all things work well.
(but program will crash after that because I change the file path.)

Please see my result in following drive:

https://goo.gl/pej1Nz

There are three output result, src image and jpn.traindata.
The jpn.traindata need to place under tesseract tessdata directory.

Regards,
Moonlove

**Comment by int3l** [comment]
Hi again,
please try to use the latest version from github.
You can install it by downloading it manually.

Check the README installation information for details.

**Comment by moonlovestar** [comment]
Hello int3l,

    I have already update my pytesseract to version 0.1.8.
    In order to clarifying this, I uploaded the log when I use 
    "pip install git+https://github.com/madmaze/pytesseract.git" in my drive.

    https://goo.gl/pej1Nz

    The filename is pytesseract_version.png.

Regards,
Moonlove

**Comment by int3l** [comment]
The version in the source code is not bumped up yet, so you are still with the old version.
Please try to use this command:

```bash
pip install -U git+https://github.com/madmaze/pytesseract.git
```

The **-U** flag is force upgrade.

**Comment by moonlovestar** [comment]
Hello int3l,

    I use -U flag to update it again but the problem still.

    Update the pytesseract_ver.png in my drive

Regards,
Moonlove

**Comment by int3l** [comment]
Very interesting.
The latest commit works as expected on my dev setup.
I also test on Win 7 SP1 x64.

Can you try to omit the --tessdata-dir config option.
This is the only difference between my example and yours.
I will also try to add this option, so we can confirm if the problem is related to it or not.

**Comment by moonlovestar** [comment]
But how can I run tesseract without set dir?
As I know, it is a bug of tesseract that cannot find tessdata dir.

Regards,
Moonlove

**Comment by int3l** [comment]
In order to still have the jpn language, you should add the **jpn.traindata** within your default Tesseract path dir - `<My Windows tessetact path>\`**tessdata**.

**Comment by moonlovestar** [comment]
Of course I have added it into my tessdata or I can even run it through directly call tesseract in command line.

Regards,
Moonlove

**Comment by int3l** [comment]
That's good.
I am just trying to reproduce your problem,

Ok, so in your test script, do you use custom config that you pass to the pytesseract method/function **image_to_string**?

**Comment by moonlovestar** [comment]
Yes, I set the tessdata  directory. I compared
    print(' '.join(command)) 
and my setting, it lost " " before send it into tesseract.
That is, the command I see is:

--tessdata-dir C:\Program Files (x86) ....

It does not have " " in my path.
I am not sure if it matters.

Regards,
Moonlove

**Comment by int3l** [comment]
Please try without the --tessdata-dir option.

**Comment by moonlovestar** [comment]
Hi int3l,

   Omitting --tessdata-dir cannot solve the problem. Please see my experiment in the top of this issue.
   It can tell you what I tried and what I found.

   My testcode has uploaded to my drive. The filename is tesseract_test.py

Regards,
Moonlove

**Comment by int3l** [comment]
Thank you, I will try to reproduce.

**Comment by int3l** [comment]
All right, you didn't mention that you are using OpenCV.
By default the **imread** method strip the alpha channel of the image...
This is the real problem. Once the alpha channel is striped - it gets replaced with black background.
(Check the documentation of OpenCV imread method for additional information)

And you are doing this explicitly before passing it to pytesseract, so there is nothing that pytesseract can do about it. That's the reason why works if you pass the input manually by modifying the code.

Basically, the workaround is to not strip the alpha channel and let pytesseract handle it, or convert the alpha channel to white color beforehand like the way pytesseract does it :)

**Comment by moonlovestar** [comment]
Hi int3l,

    It makes sense! I always forget this problem before.
    I can successfully do tesseract on my image if I use PIL to open the image instead of imread.
    You are definitively right.

    However, why you mentioned that I can either convert the alpha channel to white color to workaround the problem? Why black color cannot be used?

   Please close the issue.

Regards,
Moonlove

**Comment by int3l** [comment]
About your question - if you have black background and the characters are black - tesseract can't recognize anything, because it 's essentially like empty solid black color image :D

Also if you check the latest commit, you can see the implementation for converting alpha channel to white color :)

Closing this one as fixed.
Please feel free to reopen if you have any other questions.
